pages/_app.tsx

```diff
@@ -31,25 +32,35 @@ const aeonik = localFont({
 	]
 });
 
+type TGetLayout = NextComponentType & {getLayout: (p: ReactElement, router: NextRouter) => ReactElement}
 const WithLayout = memo(function WithLayout(props: AppProps): ReactElement {
 	const	{Component, pageProps, router} = props;
-	const	getLayout = (Component as NextComponentType & {getLayout: (p: ReactElement) => ReactElement}).getLayout || ((page: ReactElement): ReactElement => page);
+	const	getLayout = (Component as TGetLayout).getLayout || ((page: ReactElement): ReactElement => page);
+
+	const	currentApp = (
+		router.pathname.startsWith('/ycrv') ? 'yCRV' :
+			router.pathname.startsWith('/vaults') ? 'Vaults' :
+				router.pathname.startsWith('/ybribe') ? 'yBribe' :
+					'Home'
+	);
```

Comment:
nit: I'd rather have this in an external function with early returns, this is quite hard to read :)

Reply:
<img width="746" alt="Screenshot 2022-12-29 at 1 45 00 pm" src="https://user-images.githubusercontent.com/78794805/209946629-bceaad1d-175b-478f-af5d-76f383949afb.png">

```typescript
function getCurrentApp(router) {
  if (router.pathname.startsWith('/ycrv')) {
    return 'yCRV';
  }
  if (router.pathname.startsWith('/vaults')) {
    return 'Vaults';
  }
  if (router.pathname.startsWith('/ybribe')) {
    return 'yBribe';
  }
  return 'Home';
}

```

Reply:
ChatGPT didn't do too bad, I'd just destructure the pathname

<img width="773" alt="Screenshot 2022-12-29 at 1 47 13 pm" src="https://user-images.githubusercontent.com/78794805/209946883-6dc77cc5-6ab4-47ab-8160-40b85f177c64.png">

```typescript
function getCurrentApp({ pathname }) {
  if (pathname.startsWith('/ycrv')) {
    return 'yCRV';
  }
  if (pathname.startsWith('/vaults')) {
    return 'Vaults';
  }
  if (pathname.startsWith('/ybribe')) {
    return 'yBribe';
  }
  return 'Home';
}

```

Reply:
or a switch
```typescript
function getCurrentApp({ pathname }) {
  switch (pathname) {
    case '/ycrv':
      return 'yCRV';
    case '/vaults':
      return 'Vaults';
    case '/ybribe':
      return 'yBribe';
    default:
      return 'Home';
  }
}

```